core/base/CCDirector.cpp

```diff
@@ -1303,11 +1303,89 @@ void Director::createStatsLabel()
     _drawnVerticesLabel->setIgnoreContentScaleFactor(true);
     _drawnVerticesLabel->setScale(scaleFactor);
 
-    auto safeOrigin          = getSafeAreaRect().origin;
-    const int height_spacing = (int)(22 / CC_CONTENT_SCALE_FACTOR());
-    _drawnVerticesLabel->setPosition(Vec2(0, height_spacing * 2.0f) + safeOrigin);
-    _drawnBatchesLabel->setPosition(Vec2(0, height_spacing * 1.0f) + safeOrigin);
-    _FPSLabel->setPosition(Vec2(0, height_spacing * 0.0f) + safeOrigin);
+    setFPSPos();
+}
+
+// Set the FPS position on disply like the numbers on numpad
+void Director::setFPSPos(FPSPosition FPSposition)
+{
+    Vec2 _fpsPosition;
+    _FPSPosition = FPSposition;
+
+    if (_displayStats && _FPSLabel && _drawnBatchesLabel && _drawnVerticesLabel)
+    {
+        auto safeOrigin          = getSafeAreaRect().origin;
```

Comment:
the safeOrign not use?

Reply:
it was allways 0 (on my maschine)

Reply:
> the safeOrign not use?



> the safeOrign not use?

I want not remove it at the moment (maybe it is using on other platforms?

Reply:
You must consider use safeOrign, otherwise the fps label can't be display on some android high level devices. at least in my phone the safeOrigin not zero.

Reply:
The problem is: I cant test it.
Maybe you can fix it if there is a mistake (missing it)?

Reply:
It I add it it can be also wrong.

Reply:
OK, I will

Reply:
Please change this snippet also on it:  (the 'no change part' makes sense)
```
void Director::setFPSPos(FPSPosition FPSposition)
{
    if (_FPSPosition == FPSposition) // no changes!
    {
       return;
    }
    static Vec2 _fpsPosition;
    _FPSPosition = FPSposition;

    if (_displayStats && _FPSLabel && _drawnBatchesLabel && _drawnVerticesLabel)
```